clients/src/test/java/org/apache/kafka/common/network/DefaultChannelMetadataRegistry.java

```diff
@@ -22,9 +22,10 @@ public class DefaultChannelMetadataRegistry implements ChannelMetadataRegistry {
 
     @Override
     public void registerCipherInformation(final CipherInformation cipherInformation) {
-        if (this.cipherInformation != null) {
-            this.cipherInformation = cipherInformation;
+        if (cipherInformation == null) {
+            throw new IllegalArgumentException("cipherInformation cannot be null");
```

Comment:
It would be nice that the thrown exception is same to `SelectorChannelMetadataRegistry#registerCipherInformation`. Could you check that?

Reply:
I see that `SelectorChannelMetadataRegistry#registerCipherInformation` does not throw any such exception. The proposed change for `DefaultChannelMetadataRegistry` is because it is only used in running test cases, not for production. @divijvaidya @dajac You want to add anything?

Reply:
Since there are no invocation of the above method, no tests have failed in local

Reply:
agree to consistence. We can throw exception in the future if extra checks are added into production code

Reply:
Shall i remove the exception?

Reply:
> Shall i remove the exception? 

yep, I prefer this way

Reply:
Done. Please review it